backend/fetcher.py

```diff
@@ -177,6 +177,28 @@ def _is_build_id(self) -> bool:
             return False
 
     def _fetch_build_logs_from_build_id(self) -> list[dict[str, str]]:
+        """
+        Obtain build logs from a failed build by traversing
+        through the task hierarchy to the buildArch task's logs
+        """
+        koji_build = self.client.getBuild(self.build_or_task_id)
+        root_task_id = koji_build['task_id']
+        # the response of getTaskDescendents:
+        #   {'112162296': [{'arch': 'noarch', 'awaited': False...
+        task_descendants = self.client.getTaskDescendents(root_task_id)[str(root_task_id)]
+        for task_info in task_descendants:
+            if task_info['arch'] == self.arch and task_info['method'] == 'buildArch' \
+                    and task_info['state'] == 5:
+                # this is the one and only ring!
+                self.build_or_task_id = task_info['id']
+                return self._fetch_task_logs_from_task_id()
+        return []
+
+    def _fetch_complete_build_logs_from_build_id(self) -> list[dict[str, str]]:
```

Comment:
I can't seem to find calls for this method. I it intended for future use or is there some change missing from PR?